src/components/recovery/RecoveryContext/index.tsx

```diff
@@ -34,31 +37,41 @@ export function RecoveryProvider({ children }: { children: ReactNode }): ReactEl
       return
     }
 
-    return txSubscribe(TxEvent.PROCESSED, async (detail) => {
-      if (!detail.txId) {
-        return
-      }
+    // Listen to history polls (only occuring when the txHistoryTag changes)
+    const listener = dispatch(
```

Comment:
Is there a way we can move this logic inside the `useRecoveryState` hook without a listener e.g. by using the `useTxHistory` hook as a dependency to the `useAsync` call?

Having this manual `refetch` toggle seems like a workaround to me.

Reply:
I've migrated the logic and added test coverage for it in 3e63cbe13722ee56c7fed9f0fd5004d433d3356d.

Reply:
After investigating this with @usame-algan, we will revert this to a listener for now as there are a lot of race conditions involved with initially loading the recovery state: delay modifiers, chain config, Safe info and provider. All of which need to be loaded beforehand but merging the refetching logic with this causes issues with loading the initial state. We intend to revisit this when we have the capacity.